### PR TITLE
ipn/localapi: add LocalAPI support for removing all profiles

### PR DESCRIPTION
**Pull request by mihaip**
Will be used to implement a "Log out all accounts" feature.

Updates #713

### CODE REVIEW

File: ipn/ipnlocal/local.go [hunk 1]
```diff
@@ -4204,6 +4204,16 @@ func (b *LocalBackend) DeleteProfile(p ipn.ProfileID) error {
 	return b.resetForProfileChangeLockedOnEntry()
 }
 
+// DeleteProfiles removes all known profiles.
```
Comment (bradfitz) [inline ipn/ipnlocal/local.go:4207]: DeleteAllProfiles

File: ipn/ipnlocal/profiles.go [hunk 2]
```diff
@@ -349,6 +350,21 @@ func (pm *profileManager) DeleteProfile(id ipn.ProfileID) error {
...
+	metricDeleteAllProfile.Add(1)
+
+	for _, kp := range pm.knownProfiles {
+		if err := pm.store.WriteState(kp.Key, nil); err != nil {
+			return err
+		}
+	}
+	pm.NewProfile()
+	maps.Clear(pm.knownProfiles)
+	return pm.writeKnownProfiles()
```
Comment (bradfitz) [inline ipn/ipnlocal/profiles.go:365]: on failure, this leaves things is a kinda inconsistent state?

Should you try to writeKnownProfiles on your way out of the error path?

**Comment by mihaip** [comment]
Maisem and I DM'ed about this a bit, and we'll probably want something different for "Log out all accounts" (it needs to talk to control for each account to revoke the key). Moving this back to draft.

**Comment by mihaip** [comment]
Closing this, we haven't need the feature, and the useful bits were pulled into #7094

**Pull request closed without merging**